### ISSUE TITLE
MaxMessageSize property in metron agent is hardcoded

### ISSUE THREAD

**Issue by martinagalabova** [cloudfoundry/loggregator#87]
Hi!

 It seems that MaxMessageSize in metron agent is hardcoded to 4k. Unfortunately this size is too small and we would like to change it without forking the release. The problem that we're facing is the following: In a CF installation DEAs sends messages that goes to the Nats_stream_frowarder. Each message contains information about all applications currently running on that DEA (incl current state, timestamp, etc). Then the nats_stream_forwarder escapes the json file that DEA has send, and logs it. Unfortunately a DEA having 12 running apps results to a log message bigger than 4k. So if there is an ELK stack in the picture this results to an error messages in the log parsers as these logs are not a valid json files.

 To summarise - if MaxMessageSize property is configurable everyone would be able to set appropriate log message size so all messages can be logged properly, and to avoid unnecessary error messages.

Regards,
Martina

**Comment by roxtar** [comment]
Hi @martinagalabova I assume that you are talking about the MaxMessageSize in the syslog forwarder config: https://github.com/cloudfoundry/loggregator/blob/develop/jobs/metron_agent/templates/syslog_forwarder.conf.erb#L2

And you would like that to be configurable through a bosh manifest property, correct?

**Comment by martinagalabova** [comment]
Hi!

 Yes, that's the config and yes, it would be great if it's possible to change it through a bosh manifest property.

Regards,
Martina

**Comment by Jim-Campbell** [comment]
We've moved this issue into the active backlog, and it should be handled in
the next week or so.

Jim
Loggregator PM

On Mon, Oct 12, 2015 at 12:22 AM, martinagalabova <notifications@github.com>
wrote:

> Hi!
>
> Yes, that's the config and yes, it would be great if it's possible to
> change it through a bosh manifest property.
>
> Regards,
> Martina
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cloudfoundry/loggregator/issues/87#issuecomment-147299469>
> .
>



-- 
Jim Campbell
Product Manager
Pivotal Labs

**Comment by Jim-Campbell** [comment]
Fixed
